### DISCUSSION TITLE
config: can't debug tests with build_flags

### DISCUSSION THREAD

**Discussion by oleynikandrey** [fredrikaverpil/neotest-golang#177] [Configuration] [answered]
### Did you check docs and existing issues?

- [X] I have read the [documentation](https://github.com/fredrikaverpil/neotest-golang/blob/main/README.md).
- [X] I have searched the existing configuration related [discussions](https://github.com/fredrikaverpil/neotest-golang/discussions/categories/configuration).
- [X] I have updated to the latest version of Neotest.
- [X] I have updated to the latest version of neotest-golang.
- [X] I have checked the Neotest log for errors (see README for instructions on enabling it).

### Neovim version (nvim -v)

v0.10.1

### Operating system/version

MacOS 14.6.1

### Output from `:checkhealth neotest-golang`

```text
neotest-golang: require("neotest-golang.health").check()

Requirements ~
- OK Binary 'go' found on PATH: /opt/homebrew/bin/go
- OK Found go.mod file for /Users/user/repoes/xxx/my_test.go in /Users/user/repoes/xxx/go.mod
- OK Treesitter parser for go is installed
- OK neotest is available
- OK nvim-treesitter is available
- OK nio is available
- OK plenary is available

DAP (optional) ~
- OK Binary 'dlv' found on PATH: /Users/user/.local/share/nvim/mason/bin/dlv
- OK dap is available
- OK dapui is available
- OK dap-go is available
```

### Your Neotest and neotest-golang Lua setup

```Lua
return {
  {
    "nvim-neotest/neotest",
    config = function()
      require("neotest").setup({
        adapters = {
          require("neotest-golang")({
            go_test_args = { "-count=1", "-tags=integration" },
            go_list_args = { "-tags=integration" },
            dap_go_opts = {
              delve = {
                build_flags = { "-tags=integration" },
              },
            },
          }),
        },
      })
    end,
  },
}
```

### Describe the problem

Hello, I'm currently using `neotest-golang` to run and debug my tests, but I'm having some issues with debugging tests that include build flags. 
I've tried passing the `build_flags` through the `dap_go_opts` option, but it doesn't seem to have any effect. It seems like the `dap-go.debug_test()` method is not being executed when a test file includes a build flag. 
Every time I try to run a test with a build flag, I receive the following error message: 
```
Error on launch: Failed to launch: Could not launch process: Not an executable file. 
```
Could you please help me understand what might be causing this issue and how I can resolve it? I'd really appreciate your assistance in this matter.

**Comment by fredrikaverpil** [comment, marked as answer]
Hey @oleynikandrey I actually think this might be a bug. I opened #178 and it would be great if you can try that out. You can just add `branch` to your config:

```lua
return {
  {
    "fredrikaverpil/neotest-golang",
    branch = "dap-build-flags",
  }
}
```

If this doesn't help , I think we need to do some debug logging (details in the README).

**Reply by oleynikandrey** [reply]
Thank you for resolving the issue. It is working now.

**Reply by fredrikaverpil** [reply]
That's really great to hear. Thanks for reporting this.

I merged the fix into main.
